### PR TITLE
Allow to update a git application

### PR DESCRIPTION
**Pull request by jinroh**
This PR adds the possibility to upgrade an application. The version comparison between to manifest is done by a simple string comparison (no order).

The modifications of the application files is still clunky since we do not have yet a proper `Remove` method implemented. It only adds and/or override the files, without deleting removed ones.

Testing is done with a local git daemon.

### CODE REVIEW

File: pkg/apps/git.go [hunk 1]
```diff
@@ -16,103 +16,116 @@ import (
 	git "gopkg.in/src-d/go-git.v4"
 	gitObj "gopkg.in/src-d/go-git.v4/plumbing/object"
 	gitSt "gopkg.in/src-d/go-git.v4/storage/filesystem"
-	gitFS "gopkg.in/src-d/go-git.v4/utils/fs"
+	gitFS "srcd.works/go-billy.v1"
```
Comment (nono) [inline pkg/apps/git.go]: Is it normal to have this import path?

File: pkg/apps/installer_test.go [hunk 2]
```diff
@@ -13,6 +21,74 @@ import (
...
+	}
+
+	// "git daemon --reuseaddr --base-path=./ --export-all ./.git"
+	localGitCmd = exec.Command("git", "daemon", "--reuseaddr", "--base-path=./", "--export-all", "./.git")
+	localGitCmd.Dir = localGitDir
+	if out, err := localGitCmd.CombinedOutput(); err != nil {
+		fmt.Println(string(out))
+		panic(err)
+	}
+}
```
Comment (nono) [inline pkg/apps/installer_test.go]: :+1:

File: pkg/apps/git.go (hunk 1 above)
Comment (jinroh) [inline pkg/apps/git.go]: @nono I don't know why they did not use `gopkg.in`. I can use the [github address](https://github.com/src-d/go-billy) instead, if you prefer...

Comment (nono) [inline pkg/apps/git.go]: OK, we can keep it

**Pull request merged**